src/btree/bt_cursor.c

```diff
@@ -422,12 +423,13 @@ __wt_cursor_valid(WT_CURSOR_BTREE *cbt, WT_ITEM *key, uint64_t recno, bool *vali
               session, cbt->ref->page, &cbt->ref->page->pg_row[cbt->slot], cbt->tmp, true));
             key = cbt->tmp;
         }
-
-        WT_RET(__btcur_bounds_contains_key(
-          session, &cbt->iface, key, WT_RECNO_OOB, &key_out_of_bounds, NULL));
-        /* The key value pair we were trying to return weren't within the given bounds. */
-        if (key_out_of_bounds)
-            return (0);
+        if (check_bounds) {
```

Comment:
This comment is addressing line 384. Would this call to `__btcur_bounds_contains_key `also require a `check_bounds` guard?

Reply:
We'll still add the check_bounds guard as that way its consistent.

Reply:
[6096b21](https://github.com/wiredtiger/wiredtiger/pull/8309/commits/6096b21b1f569d91c83d5af8d2fafce7effe577b)